AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -166,8 +177,28 @@ - (void)enqueueItem:(id<MSLog>)item flags:(MSFlags)flags {
         [self checkPendingLogs];
       }
     }
+  };
+#if !TARGET_OS_OSX
+
+  // Process synchronously in case applicationWillTerminate was hit.
+  (self.applicationWillTerminateEntered ? dispatch_sync : dispatch_async)
```

Comment:
Wait. `dispatch_sync` can be called not from the main thread. The execution time is the user's responsibility in this case?

Reply:
Yes we don't control the context in this case it's whatever caller context is running. Need to be careful.

Reply:
The more I am thinking about this the more I am thinking we should postpone this fix. It comes with some risks at this point.